packages/agent-utils/src/embeddings/LocalVectorDB.ts

```diff
@@ -0,0 +1,50 @@
+import { EmbeddingApi } from "./EmbeddingApi";
+import { normalize, normalizedCosineSimilarity } from "./utils";
+import { LocalDocumentStore } from "./LocalDocumentStore";
+import { LocalDocument } from "./LocalDocument";
+
+export class LocalVectorDB {
+  constructor(
+    private embeddingApi: EmbeddingApi,
+    private store: LocalDocumentStore,
+  ) {}
+
+  async add(items: {
+    text: string
+  }[]): Promise<void> {
+    const itemTexts = items.map(item => item.text);
+    const results = await this.embeddingApi.createEmbeddings(itemTexts)
+
+    for await (const result of results) {
+      this.store.add({
+        text: result.input,
+        vector: result.embedding,
+      })
+    }
+  }
+
+  async search(query: string, limit: number): Promise<LocalDocument[]> {
+    const queryEmbeddingResults = await this.embeddingApi.createEmbeddings(query);
+    const queryVector = queryEmbeddingResults[0].embedding;
+    const normalizedQueryVector = normalize(queryVector);
+
+    const documents = this.store.list();
+
+    const distances = documents.map(document => {
```

Comment:
If memory ever becomes an issue, we could do this for every item and just store a ref to document and distance into memory, instead of all the documents  (like we do above with `.list`)